lib/src/migrators/module.dart

```diff
@@ -318,64 +319,70 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
   @override
   void visitStylesheet(Stylesheet node) {
     var oldNamespaces = _namespaces;
-    var oldHasUseRule = _hasUseRule;
+    var oldHasUseRule = _usedUrls;
     var oldAdditionalUseRules = _additionalUseRules;
     var oldUseAllowed = _useAllowed;
-    _namespaces = {};
-    _hasUseRule = {};
+    _usedUrls = {};
     _additionalUseRules = {};
     _useAllowed = true;
-    _determineNamespaces(node.span.sourceUrl);
+    _namespaces = _determineNamespaces(node.span.sourceUrl);
     super.visitStylesheet(node);
     _namespaces = oldNamespaces;
-    _hasUseRule = oldHasUseRule;
+    _usedUrls = oldHasUseRule;
     _additionalUseRules = oldAdditionalUseRules;
     _lastUrl = node.span.sourceUrl;
     _useAllowed = oldUseAllowed;
   }
 
   /// Determines namespaces for all `@use` rules that the stylesheet will
   /// contain after migration.
-  void _determineNamespaces(Uri url) {
+  Map<Uri, String> _determineNamespaces(Uri url) {
+    var namespaces = <Uri, String>{};
     var sourcesByNamespace = <String, Set<ReferenceSource>>{};
-    for (var reference in references.referenceSources.keys) {
-      if (reference.span.sourceUrl == url) {
-        var source = references.referenceSources[reference];
-        var namespace = source.defaultNamespace;
-        if (namespace != null) {
-          sourcesByNamespace.putIfAbsent(namespace, () => {}).add(source);
-        }
+    for (var reference in references.sources.keys) {
+      if (reference.span.sourceUrl != url) continue;
+      var source = references.sources[reference];
+      var namespace = source.defaultNamespace;
+      if (namespace == null) continue;
+      // Existing `@use` rules should always keep their namespaces.
```

Comment:
Style nit: a newline above here would help visually differentiate the "determining which sources to add" step from the "adding those sources" step.

Reply:
Done